Build error due to to type conversion

After upgrading to torch==1.7.0 (Ubuntu 16.04) I was unable to install torch-blocksparse due to error during building 'torch_blocksparse_cpp_utils' extension:

```
Building wheels for collected packages: torch-blocksparse
  Building wheel for torch-blocksparse (setup.py) ... error

  ERROR: Command errored out with exit status 1:
...

  Complete output (43 lines):
  running bdist_wheel
  /home/superuser/khovrichev/ru-gpts/gpt_env/lib/python3.7/site-packages/torch/utils/cpp_extension.py:339: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
    warnings.warn(msg.format('we could not find ninja.'))
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/deepspeedsparseselfattention.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/__init__.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/matmul.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/conv.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/scratch.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/attention.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/permute.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/batchnorm.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/softmax.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  copying torch_blocksparse/relu.py -> build/lib.linux-x86_64-3.7/torch_blocksparse
  running build_ext
  building 'torch_blocksparse_cpp_utils' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/csrc

gcc -pthread -B /home/superuser/anaconda3/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/home/superuser/khovrichev/ru-gpts/gpt_env/lib/python3.7/site-packages/torch/include -I/home/superuser/khovrichev/ru-gpts/gpt_env/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/superuser/khovrichev/ru-gpts/gpt_env/lib/python3.7/site-packages/torch/include/TH -I/home/superuser/khovrichev/ru-gpts/gpt_env/lib/python3.7/site-packages/torch/include/THC -I/home/superuser/khovrichev/ru-gpts/gpt_env/include -I/home/superuser/anaconda3/include/python3.7m -c csrc/utils.cpp -o build/temp.linux-x86_64-3.7/csrc/utils.o -O2 -fopenmp -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=torch_blocksparse_cpp_utils -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14

  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  csrc/utils.cpp: In function ‘void segment_blocks(at::Tensor, at::Tensor, at::Tensor, int, ret_t&)’:
  csrc/utils.cpp:48:37: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           for(int nn = n_left + 1; nn < n; nn++)
                                       ^
  csrc/utils.cpp:68:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           for(size_t km = 0; km < max_width; km++)
                                 ^
  csrc/utils.cpp:69:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           for(size_t kn = 0; kn < max_width; kn++)

                                 ^
csrc/utils.cpp:91:50: error: converting to ‘std::vector<std::tuple<int, at::Tensor> >::value_type {aka std::tuple<int, at::Tensor>}’ from initializer list would use explicit constructor ‘constexpr std::tuple<_T1, _T2>::tuple(_U1&&, _U2&&) [with _U1 = int&; _U2 = at::Tensor; <template-parameter-2-3> = void; _T1 = int; _T2 = at::Tensor]’
       ret.push_back({max_width, torch::cat(to_cat)});

                                                    ^
  csrc/utils.cpp: In function ‘ret_t sdd_segment(at::Tensor, int)’:
  csrc/utils.cpp:115:88: warning: narrowing conversion of ‘H’ from ‘size_t {aka long unsigned int}’ to ‘long int’ inside { } [-Wnarrowing]
     torch::Tensor scratch = torch::empty({H, layout.sum().item<int>(), 4}, layout.dtype());
                                                                                          ^
  csrc/utils.cpp:115:88: warning: narrowing conversion of ‘H’ from ‘size_t {aka long unsigned int}’ to ‘long int’ inside { } [-Wnarrowing]
  error: command 'gcc' failed with exit status 1
```

I am not aware if this is compatibility problem or something related to gcc version.
The possible way to fix this is replace explicit tuple constructor with `std::make_tuple` factory here:  (PR #30)

https://github.com/ptillet/torch-blocksparse/blob/9aa94789f13ada713af36cfd8cca2fc9a7f6b79a/csrc/utils.cpp#L91

--- a/csrc/utils.cpp
+++ b/csrc/utils.cpp
@@ -88,7 +88,7 @@ void segment_blocks(torch::Tensor layout, torch::Tensor idx, torch::Tensor scrat
     if(current[h] > 0)
       to_cat.push_back(scratch[h].slice(0, 0, current[h]));
   if(!to_cat.empty())
-    ret.push_back({max_width, torch::cat(to_cat)});
+    ret.push_back(std::make_tuple(max_width, torch::cat(to_cat)));
 }
 
 

